src/timeseriesflattener/flattened_dataset.py

```diff
@@ -27,69 +27,105 @@ def add_outcome(
         self,
         outcome_df: DataFrame,
         lookahead_days: float,
-        resolve_multiple: str,
-        fallback: List[str],
-        values_colname: str = "values",
+        resolve_multiple: Callable,
+        fallback: float,
+        source_values_colname: str = "val",
         new_col_name: str = None,
     ):
         """Adds an outcome-column to the dataset
 
         Args:
             outcome_df (DataFrame): Cols: dw_ek_borger, datotid, value if relevant.
             lookahead_days (float): How far ahead to look for an outcome in days. If none found, use fallback.
-            resolve_multiple (str): What to do with more than one value within the lookahead.
-                Suggestions: earliest, latest, mean, max, min.
-            fallback (List[str]): What to do if no value within the lookahead.
-                Suggestions: latest, mean_of_patient, mean_of_population, hardcode (qualified guess)
-            values_colname (str): Colname for outcome values in outcome_df
+            resolve_multiple (Callable): How to handle multiple values within the lookahead window. Takes a a function that takes a list as an argument and returns a float.
+            fallback (float): What to do if no value within the lookahead.
+            source_values_colname (str): Colname for outcome values in outcome_df. Defaults to "val".
```

Comment:
Not entirely sure what this argument values. Is it the column containing values? (assuming there is only one column that does this in a long format style?) or is it the column of outcomes?

Reply:
"Long format style"? 

Currently, the values must be floats or ints – any handling of e.g. strings has to happen before flattening. 

I've expanded the docstring.

Reply:
[long format and wide format](https://www.statology.org/long-vs-wide-data/) are two ways to represent data. Long format is special because it forces only one column with values.